### ISSUE TITLE
Zero child of CanvasLayer (which is a child of Camera2D) may render with wrong transform [Windows, Intel HD 630 or lower]

### ISSUE THREAD

**Issue by arkology** [godotengine/godot#58314]
___

***Bugsquad note:** This issue has been confirmed several times already. No need to confirm it further.*

___

### Godot version

4.0 alpha2 (UPD: alpha5)

### System information

1) Windows 10, Dell Inspiron 7577, Intel HD Graphics 630 (driver version (latest atm) 27.20.100.9664), GTX 1050ti (driver version 30.0.14.7212), Vulkan API 1.2.170
2) PC from my work with Intel HD Graphics 620 and Windows 10

### Issue description

CanvasLayer, being Camera2D child, doesn't "follow" Camera2D if global (0,0) coordinate is in view. "Follow" is not properly word here, but I don't know how to describe, so just look at gifs below. NOTE: It works as expected (like on second gif) in 3.x, so this actually a bug and not wrong use case.
MRP scene tree:
![Godot_v4 0-alpha2_win64_ESlZE3zWcQ](https://user-images.githubusercontent.com/43543909/154797687-5bd3b812-964e-496a-b2d9-36dfc9db7d5b.png)

Broken when running on Intel HD Graphics 630:
<details>
  <summary>Gif</summary>

![RICEamGHyY](https://user-images.githubusercontent.com/43543909/154797395-93817510-48f8-4f48-bd6f-2b844590db64.gif)

</details>

Works as expected on 1050ti:
<details>
  <summary>Gif</summary>

![mNvK53vSxB](https://user-images.githubusercontent.com/43543909/154797753-720b7ece-5815-4a26-b8af-358ab5ce985b.gif)

</details>

Console output is the same and doesn't report anything strange.

If this is driver or hardware issue, I don't understand why it works if global (0,0) coordinate is out of the screen.
Also for Intel HD Graphics 630 this is latest driver version (according to Dell update center).
Actually this could be Intel iGPU problem, see #59802

### Steps to reproduce

Run MRP (on Intel iGPU & Windows10)

### Minimal reproduction project

[G4CanvasLayerBug.zip](https://github.com/godotengine/godot/files/8101961/G4CanvasLayerBug.zip)

**Comment by TheDuriel** [comment]
This is not a bug, but a wrong use case. CanvasLayers should not change their properties based on any camera of viewport properties, unless they are parallax layers, or your own code does so.

The purpose of a canvaslayer is to separate something from world space, and to render it always on top or below of the implicit layer 0.

No matter where in your tree, canvas layer 1 will always display on top of your world. And on top of whatever your camera sees within that world.

**Comment by arkology** [comment]
> This is not a bug, but a wrong use case. CanvasLayers should not change their properties based on any camera of viewport properties, unless they are parallax layers, or your own code does so.
> 
> The purpose of a canvaslayer is to separate something from world space, and to render it always on top or below of the implicit layer 0.
> 
> No matter where in your tree, canvas layer 1 will always display on top of your world. And on top of whatever your camera sees within that world.

This is not wrong use case. Because it works as expected in 3.x. Look at gifs. CanvasLayer doesn't follow Camera2D sometimes.

**Comment by TheDuriel** [comment]
The only case in which the behavior you desire should happen, is if you enable https://docs.godotengine.org/en/stable/classes/class_canvaslayer.html#class-canvaslayer-property-follow-viewport-enable this property.

**Comment by arkology** [comment]
> The only case in which the behavior you desire should happen, is if you enable https://docs.godotengine.org/en/stable/classes/class_canvaslayer.html#class-canvaslayer-property-follow-viewport-enable this property.

Nope. I found this porting my project from 3.4.2 to 4.0. All properties are the same.

**Comment by arkology** [comment]
Godot 3.4.2. Same setup. Works as expected.
![jgUfKWvgTJ](https://user-images.githubusercontent.com/43543909/154804885-707f0f3c-19cf-4061-a42c-a51f820d06f9.gif)

[G3CanvasLayerProve.zip](https://github.com/godotengine/godot/files/8102335/G3CanvasLayerProve.zip)

**Comment by arkology** [comment]
Looks like I found what is wrong. I think it may be caused by https://github.com/godotengine/godot/commit/c1023157eb761efcda07c3fa2015517b8a3621e1
Now I'll try to dig deeper and create project showing it

**Comment by arkology** [comment]
Almost same project, but scene tree modified a bit:
![Godot_v4 0-alpha2_win64_DijzLU3bGv](https://user-images.githubusercontent.com/43543909/154806529-0db08a39-f707-4961-a05a-6289f0581a85.png)
Look at NinePatchRect order. The position of each next NinePatchRectX is to the right of the previous one.
This project running:
![HJdmMcnp2n](https://user-images.githubusercontent.com/43543909/154806574-017be06b-1a3a-4c9f-bd71-8391f0d15363.gif)
NinePatchRect node behaves weird.

Lets change nodes order it tree (without changing position). NinePatchRect2 first:
![Godot_v4 0-alpha2_win64_AUgv9ZVZ3r](https://user-images.githubusercontent.com/43543909/154806683-50b0e0ec-539f-47f1-b5ae-d8edb57c8f8d.png)
Run this setup:
![MDne8OpGT1](https://user-images.githubusercontent.com/43543909/154806710-71aba984-18c5-4cbe-894c-75be158d8286.gif)
NinePatchRect2 node behaves weird.

And so on. This is not wrong use case, this is wrong behaviour.

MRP: 
[G4CanvasLayerBug.zip](https://github.com/godotengine/godot/files/8102451/G4CanvasLayerBug.zip)

I think it could be caused by https://github.com/godotengine/godot/commit/c1023157eb761efcda07c3fa2015517b8a3621e1
@KoBeWi please take a look.

**Comment by arkology** [comment]
Looks like usage of "gl_compatibility" renderer solves the problem (thought not properly tested yet) as it was mentioned in https://github.com/godotengine/godot/issues/71027

**Comment by Skyseb** [comment]
I have same problem

Godot Engine v4.3.beta1.official.a4f2ea91a - https://godotengine.org
Vulkan 1.2.137 - Forward+ - Using Device #0: Intel - Intel(R) HD Graphics 630

![Control1](https://github.com/godotengine/godot/assets/56416214/67ca1adf-dac7-43a1-9882-837bc50277ec)

first element wrong .. 
![L](https://github.com/godotengine/godot/assets/56416214/25565a6d-1dff-46c5-acd2-a1ea26acdd4d)

if change.. to label.

![L](https://github.com/godotengine/godot/assets/56416214/276d8752-347a-4c0f-9fdd-669ec1db04f9)

only "L" .. woow... 

Minimal Proyect:
[MyPlatform4.3.zip](https://github.com/user-attachments/files/15893619/MyPlatform4.3.zip)

**Comment by MajorH5** [comment]
Just for anyone else who encounters this in the future, a work around I used is to force Godot to render the first child as any blank texture. Ex. before adding any children to the canvas layer, add a label with a single space as text. All children following that label should appear as normal.

**Comment by arkology** [comment]
I decided to test this again and I can no longer reproduce the problem on the computer I have.
`Godot v4.4.dev7 - Windows 10 (build 19041) - Single-window, 2 monitors - Vulkan (Forward+) - integrated Intel(R) HD Graphics 620 (Intel Corporation; 26.20.100.7157) - Intel(R) Core(TM) i3-7130U CPU @ 2.70GHz (4 threads)`

And unfortunately I don't know what changed - something on drivers side or on Godot side (last time I tried - 2D batching changed how it will break (looked differently), but it was still broken).
Would be great if someone will also re-test if this still reproducible.

**Comment by arkology** [comment]
Okay, since no one responded and this issue has been removed from 4.x milestone, I just close it (although the bug remains in 4.0-4.3).
